main.js

```diff
@@ -44,34 +55,84 @@ function draw() {
     points.push([x, y]);
   }
 
-  const minX = Math.min(...points.map((p) => p[0]));
-  const maxX = Math.max(...points.map((p) => p[0]));
-  const minY = Math.min(...points.map((p) => p[1]));
-  const maxY = Math.max(...points.map((p) => p[1]));
+  return points;
+}
+
+/**
+ *
+ * @param {HTMLCanvasElement} canvas
+ * @param {number[][]} points
+ */
+function drawPoints(canvas, points) {
+  let minX = Infinity,
+    maxX = -Infinity,
+    minY = Infinity,
+    maxY = -Infinity;
+
+  for (const point of points) {
+    const x = point[0];
+    const y = point[1];
+
+    minX = Math.min(x, minX);
+    maxX = Math.max(x, maxX);
+    minY = Math.min(y, minY);
+    maxY = Math.max(y, maxY);
+  }
+
+  const context = canvas.getContext("2d");
+  const shapeWidth = maxX - minX;
+  const shapeHeight = maxY - minY;
+  const shapeMidX = (maxX + minX) / 2;
+  const shapeMidY = (maxY + minY) / 2;
+  const viewportWidth = canvas.width - config.margin * 2;
+  const viewportHeight = canvas.height - config.margin * 2;
 
+  context.save();
+
+  // A bunch of transformations, so that the the rendering looks centered and scaled to fit viewport
```

Comment:
Added this to support rendering very big outputs and centering it nicely with margins